bench/tx-generator/src/Cardano/TxGenerator/Setup/NixService.hs

```diff
@@ -42,31 +53,76 @@ data NixServiceOptions = NixServiceOptions {
   , _nix_init_cooldown    :: Double
   , _nix_era              :: AnyCardanoEra
   , _nix_plutus           :: Maybe TxGenPlutusParams
+  , _nix_keepalive        :: Maybe Integer
   , _nix_nodeConfigFile       :: Maybe FilePath
   , _nix_cardanoTracerSocket  :: Maybe FilePath
   , _nix_sigKey               :: SigningKeyFile In
   , _nix_localNodeSocketPath  :: String
-  , _nix_targetNodes          :: NonEmpty NodeIPv4Address
+  , _nix_targetNodes          :: NonEmpty NodeDescription
   } deriving (Show, Eq)
 
 deriving instance Generic NixServiceOptions
 
+-- only works on JSON Object types
+data NodeDescription =
+  NodeDescription {
+      -- NodeIPAddress would be agnostic to IPv4 vs. IPv6 and likely
+      -- a small investment here.
+      ndAddr    :: NodeIPv4Address
+    , ndName    :: String
+    } deriving (Eq, Show, Generic)
+
+-- { "alias": "foo", "addr": ..., "port": ... }
+instance FromJSON NodeDescription where
+  parseJSON = withObject "NodeDescription" \v -> do
+    unNodeHostIPv4Address
+            <- v .:  "addr"    <?> Key "addr"
+    naPort  <- fmap toEnum $
+                 v .:  "port"  <?> Key "port"
+    let naHostAddress = NodeHostIPv4Address {..}
+        ndAddr        = NodeAddress {..}
+    ndName  <- v .:? "name"    <?> Key "name" .!= show ndAddr
+    pure $ NodeDescription {..}
+
+instance ToJSON NodeDescription where
+  toJSON NodeDescription {..} = object
+       [ "name" .= ndName
+       , "addr" .= unNodeHostIPv4Address
+       , "port" .= fromEnum naPort ] where
+    _addr@NodeAddress {..} = ndAddr
+    _hostAddr@NodeHostIPv4Address {..} = naHostAddress
+
+
+-- Long GC pauses on target nodes can trigger spurious MVar deadlock
+-- detection. Increasing this timeout can help mitigate those errors.
+-- 10s turned out to be a problem, so it's 30s now.
+defaultKeepaliveTimeout :: Clock.DiffTime
+defaultKeepaliveTimeout = 30
+
+getKeepaliveTimeout :: NixServiceOptions -> Clock.DiffTime
+getKeepaliveTimeout = maybe defaultKeepaliveTimeout Clock.secondsToDiffTime . _nix_keepalive
+
+getKeepaliveTimeout' :: Maybe NixServiceOptions -> Clock.DiffTime
+getKeepaliveTimeout' = maybe defaultKeepaliveTimeout getKeepaliveTimeout
+
```

Comment:
Could you merge those two into a single function?

Reply:
I also folded this back into the commit touching these lines in a rebase.